elfi/methods/parameter_inference.py

```diff
@@ -1050,19 +1048,26 @@ def sample(self, n_samples, warmup=None, n_chains=4, threshold=None, initials=No
                 raise ValueError("The shape of initials must be (n_chains, n_params).")
         else:
             # TODO: now GPy specific
-            inds = np.argsort(self.target_model._gp.Y[:,0])[:n_chains]
+            inds = np.argsort(self.target_model._gp.Y[:,0])
             initials = np.asarray(self.target_model._gp.X[inds])
 
         self.target_model.is_sampling = True  # enables caching for default RBF kernel
 
         random_state = np.random.RandomState(self.seed)
         tasks_ids = []
+        ii_initial = 0
 
         # sampling is embarrassingly parallel, so depending on self.client this may parallelize
         for ii in range(n_chains):
             seed = get_sub_seed(random_state, ii)
-            tasks_ids.append(self.client.apply(mcmc.nuts, n_samples, initials[ii], posterior.logpdf,
+            while np.isinf(posterior.logpdf(initials[ii_initial])):  # discard bad initialization points
+                ii_initial += 1
+                if ii_initial == len(inds):
+                    raise ValueError("Cannot find enough initialization points!")
```

Comment:
Hmm. wouldn't it make sense to even start from the same place rather than just throw error in these cases?